### PR TITLE
fix: sonarcloud coverage report [SV-70]

### PR DESCRIPTION
**Pull request by FRSgit**
### 🔗 Linked issue

https://vsf.atlassian.net/browse/SV-70

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSDoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

add yarn install step to generate .nuxt tsconfig before check before sonarcloud analysis
migrate to the new default vitest coverage reporter

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [] I have updated the documentation accordingly.

### CODE REVIEW

**Review by dawid-ziobro — APPROVED** [review]

File: apps/web/package.json [hunk 1]
```diff
@@ -34,6 +34,7 @@
     "@types/lodash-es": "^4.17.7",
     "@types/node": "^18",
     "@vite-pwa/nuxt": "^0.1.0",
+    "@vitest/coverage-v8": "^0.33.0",
```
Comment (Szymon-dziewonski) [inline apps/web/package.json:37]: question: why is it added?

File: .github/workflows/code-quality.yaml [hunk 2]
```diff
@@ -26,6 +27,16 @@ jobs:
...
           fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: './.github/.node-version'
+          cache: 'yarn'
+
+      - name: Install dependencies
+        run: yarn --frozen-lockfile
```
Comment (Szymon-dziewonski) [inline .github/workflows/code-quality.yaml:38]: what does this change? after it you do not build anything

Comment (Szymon-dziewonski) [reply .github/workflows/code-quality.yaml:38]: okay I see there is `    "postinstall": "nuxt prepare",` then its clear now

File: apps/web/package.json (hunk 1 above)
Comment (FRSgit) [reply apps/web/package.json:37]: in the description "migrate to the new default vitest coverage reporter" - there is a warning being thrown when running our current coverage script that vitest has changed default reporter and you need to install this package. I've tested and coverage works, so 🤷 
They basically remove whole `@vitest/coverage-c8` package: https://github.com/vitest-dev/vitest/commit/a90d64fee3d07037ffccac54c032603cca00d10a

I've found the motivation behind this change in this PR: https://github.com/vitest-dev/vitest/pull/3339

Comment (Szymon-dziewonski) [reply apps/web/package.json:37]: yeah description does not describe why its added, migration means usually to install and use, in this case you just installed package, and thats it. Therefor my question. So if I understand correctly vitest pick it up and use internally?

Comment (FRSgit) [reply apps/web/package.json:37]: yea, vitest up to now has used `@vitest/coverage-c8` as the default coverage reported (it probably wasn't installed by mistake the coverage script in our repo does not work without any reporter).
And today, when I've tried to run the coverage script I've got this:
image

Comment (Szymon-dziewonski) [reply apps/web/package.json:37]: okay, thanks for explanation :)

**Review by Szymon-dziewonski — APPROVED** [review]

**Pull request merged**